hedera-node/hedera-admin-service-impl/src/main/java/com/hedera/node/app/service/admin/impl/handlers/FreezeHandler.java

```diff
@@ -107,25 +119,70 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         // it must be submitted by an account with superuser privileges, that is checked during ingest
     }
 
+    public void handle(
+            @NonNull TransactionBody txn,
+            @NonNull final AdminServiceConfig adminServiceConfig,
+            @NonNull WritableSpecialFileStore specialFileStore,
+            @NonNull SwirldDualState dualState) {
+        final FreezeUpgradeActions upgradeActions =
+                new FreezeUpgradeActions(adminServiceConfig, dualState, specialFileStore);
+
+        requireNonNull(txn);
+        requireNonNull(adminServiceConfig);
+        requireNonNull(specialFileStore);
+        // TODO: requireNonNull(dualState);
+        // for the time being, this will always be null because we are not using SwirldDualState
```

Comment:
Why does this have to be null? Shouldn't it be non-null, and the tests would be sure to mock out the instance before calling? I guess we don't have the handle workflow yet anyway so there is no other way for this method to be called except by tests.